### PR TITLE
oracle-instant-client: add macOS conflict

### PR DESCRIPTION
**Pull request by adamjstewart**
Doesn't have M1 support yet so I didn't bother to add downloads/installs for macOS, but hopefully someday we will have them.

### CODE REVIEW

File: var/spack/repos/builtin/packages/oracle-instant-client/package.py [hunk 1]
```diff
@@ -57,6 +57,10 @@ class OracleInstantClient(Package):
 
     depends_on('libaio', type='link')
 
+    # TODO: add URLs for macOS. Unfortunately still no native M1 support.
+    # https://www.oracle.com/database/technologies/instant-client/macos-intel-x86-downloads.html
+    conflicts('platform=darwin')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/oracle-instant-client/package.py:62]:
```suggestion
    conflicts('platform=darwin target=aarch64:')
```

**Review by alalazo — CHANGES_REQUESTED** [review]
Does it make sense to restrict the constraint as in the suggestion?

File: var/spack/repos/builtin/packages/oracle-instant-client/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/oracle-instant-client/package.py:62]: Until someone adds URLs for macOS x86_64, x86_64 and aarch64 are both not supported.

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
@spackbot run pipeline

**Pull request merged**